lib/common/json-schema-validator.js

```diff
@@ -0,0 +1,51 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+module.exports = jsonSchemaValidatorFactory;
+
+jsonSchemaValidatorFactory.$provide = 'JsonSchemaValidator';
+jsonSchemaValidatorFactory.$inject = [
+    'ajv'
+];
+
+function jsonSchemaValidatorFactory(
+    Ajv
+) {
+    function JsonSchemaValidator () {
+        this._ajv = new Ajv({ allErrors: true,  verbose: true});
```

Comment:
Agreed, Since this is base class, option 2 maybe better.  The specific option would be inside the sub class